connectors/bigquery/flink-connector-gcp-bigquery/src/main/java/com/google/flink/connector/gcp/bigquery/BigQueryCatalog.java

```diff
@@ -0,0 +1,396 @@
+                return List.of();
+            }
+            datasets
+                    .iterateAll()
+                    .forEach(
+                            dataset -> targetReturnList.add(String.format("%s", dataset.getDatasetId().getDataset())));
+            return targetReturnList;
+        } catch (Exception e) {
+            throw new CatalogException("Failed to list databases", e);
+        }
+
+    }
+
+    @Override
+    public CatalogDatabase getDatabase(String databaseName) throws DatabaseNotExistException, CatalogException {
+        DatasetId datasetId = DatasetId.of(this.projectId, databaseName);
+        Dataset dataset = bigqueryclient.client.getDataset(datasetId);
+        if (dataset == null) {
+            throw new DatabaseNotExistException(getName(), databaseName);
+        }
+
+        return new BigQueryCatalogDatabase(this.projectId, dataset);
+    }
+
+    @Override
+    public boolean databaseExists(String databaseName) throws CatalogException {
+        try {
+            Page<Dataset> datasets = bigqueryclient.client.listDatasets(this.projectId);
+            if (datasets != null) {
+                for (Dataset dataset : datasets.iterateAll()) {
+                    if (dataset.getDatasetId().getDataset().equals(databaseName)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        } catch (BigQueryException e) {
+            throw new CatalogException("Failed to check if database exists: " + databaseName, e);
+        }
+    }
+
+    @Override
+    public void createDatabase(String databaseName, CatalogDatabase database, boolean ignoreIfExists) throws DatabaseAlreadyExistException, CatalogException {
+        throw new UnsupportedOperationException("Function createDatabase not supported yet.");
+    }
+
+    @Override
+    public void dropDatabase(String databaseName, boolean ignoreIfNotExists, boolean cascade) throws DatabaseNotExistException, DatabaseNotEmptyException, CatalogException {
+        throw new UnsupportedOperationException("Function dropDatabase not supported yet.");
+    }
+
+    @Override
+    public void alterDatabase(String name, CatalogDatabase newDatabase, boolean ignoreIfNotExists) throws DatabaseNotExistException, CatalogException {
+        throw new UnsupportedOperationException("Function alterDatabase not supported yet.");
+    }
+
+    // -- Table Operations --
+    @Override
+    public List<String> listTables(String databaseName) throws DatabaseNotExistException, CatalogException {
+        List<String> targetReturnList = new ArrayList<>();
+        try {
+            DatasetId datasetId = DatasetId.of(this.projectId, databaseName);
+            Page<Table> tables = bigqueryclient.client.listTables(datasetId, TableListOption.pageSize(100));
+
+            tables.iterateAll().forEach(table -> targetReturnList.add(String.format("%s, (Type: %s)",
```

Comment:
Should probably filter for tables only (like you do below with view).

Reply:
Also should probably also be the tablename only since a user may want to iterate all tables in a space programmatically.